src/App.tsx

```diff
@@ -15,48 +16,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET: return good1.localeCompare(good2);
+
+      case SortType.LENGTH: return good1.length - good2.length;
+
+      default: return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsRevesed] = useState(false);
+  const visibleGoods = getReorderedGoods(
+    goodsFromServer, { sortType, isReversed },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames('button is-info',
+            { 'is-light': sortType !== 1 })}
+          onClick={() => setSortType(SortType.ALPHABET)}
```

Comment:
You should create a separate function for each handle click.

Reply:
It is reasonable